.github/workflows/ruff.yml

```diff
@@ -13,10 +13,14 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v4
-
-    - run: python -Im pip install --user ruff
-
-    - name: Run ruff
-      working-directory: ./src
-      run: ruff --output-format=github wagtailmarkdown
+      - uses: actions/checkout@v4
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install ruff
+      - name: Run Ruff
+        run: ruff check --output-format=github ./src/wagtailmarkdown
```

Comment:
These changes are not needed.

What we do need is to install the same ruff version as in pre-commit https://github.com/torchbox/wagtail-markdown/blob/main/.pre-commit-config.yaml#L23C11-L23C17

a la https://github.com/torchbox/wagtail-periodic-review/blob/main/.github/workflows/ruff.yml